packages/x-telemetry/README.md

```diff
@@ -0,0 +1,70 @@
+# @mui/x-telemetry
+
+Package used by some of MUI X to collects **completely anonymous** temetry data about general usage. Participation in this anonymous program is optional, and you may opt-out if you'd not like to share any information.
+
+## How to opt-in
+
+Currently, **it's disabled by default,** and you could opt-in to it in 3 ways:
+1. By using CLI util to enable/disable it for the current machine.
+```shell
+npx mui-x-telemetry enable
+```
+
+2. By setting the flag to global object on the application start (e.g. in main file).
+```js
+import { ponyfillGlobal } from "@mui/utils";
+
+ponyfillGlobal.__MUI_X_TELEMETRY_ENABLED__ = true;
+// or
+ponyfillGlobal.__MUI_X_TELEMETRY_DISABLED__ = true;
```

Comment:
Instead of exposing `ponyfillGlobal`, I would prefer exporting a function from `@mui/x-telemetry`:
```ts
import { setTelemetryStatus } from "@mui/x-telemetry";

setTelemetryStatus(true);
// or
setTelemetryStatus(false);
```

WDYT?

Reply:
Sure, we would have this ability, which I've also documented in the readme file.
```js
import { muiXTelemetrySettings } from '@mui/x-telemetry';

muiXTelemetrySettings.enableTelemetry();
// or
muiXTelemetrySettings.disableTelemetry();
```

The main problem why I've added the option with setting on the global object, is that in some cases when you would install only `@mui/x-data-grid-pro,` some package managers (as `pnpm`) would not allow you to access `@mui/x-telemetry` package without adding it to your package.json:
 
<img src="https://github.com/mui/mui-x/assets/12295460/6a1810c3-f7b2-4f19-a403-4a4f170dd459" width="400">

Reply:
In the next major version, people using a commercial package will have to set their license using `@mui/x-license`, they won't be able to do it through the commercial package.
And as you said they will have to install `@mui/x-license` explicitly when using pnpm.

So if you need to expose something, you can always re-export it from `@mui/x-license` :+1: 
As long as the name is explicit enough outside of the telemetry package.

Reply:
`@mui/x-telemetry` being a dep of `@mui/x-license` in this package and not a dep of the other packages, I took for granted that it was only targeting commercial users.
But if this is not the case then you are entirely right.

Reply:
> @mui/x-telemetry being a dep of @mui/x-license in this package and not a dep of the other packages, I took for granted that it was only targeting commercial users.

I didn't notice this and assumed that we were targeting all users 😅
Anyway, it's good that this question came up!
@joserodolfofreitas What do you think? Should we only target commercial users?

Reply:
I think we should start targeting commercial users. In the future, it'd also be nice to collect data from the community plan, although it may be harder to convince our users to do that. 
With the license validation there's almost an expectation that there's a connection with our servers at some point.